checker/manual/external-tools.tex

```diff
@@ -444,37 +444,177 @@ \section{Maven\label{maven}}
-      '-implicit:class',
-      '-processor', 'org.checkerframework.checker.nullness.NullnessChecker'
-    ]
+          // uncomment to turn Checker Framework errors into warnings
+          // '-Awarns',
+          '-processorpath', "${configurations.checkerFramework.asPath}",
+          '-processor', 'org.checkerframework.checker.nullness.NullnessChecker',
+          "-Xbootclasspath/p:${configurations.checkerFrameworkAnnotatedJDK.asPath}"
+        ]
+  }
+}
+\end{Verbatim}
+\end{enumerate}
+
+\subsection{Gradle with JDK 7\label{gradle-jdk-7}}
+These instructions explain how to configure Gradle to use the Checker Framework using JDK 7.
+See Section~\ref{gradle-jdk-8} for JDK 8 instructions.
+
+\begin{enumerate}
+
+\item Define the Maven Central repostiory:
+
+\begin{Verbatim}
+repositories {
+  ... extisting repositories...
+  mavenCentral()
+}
+\end{Verbatim}
+
+\item Add dependency configurations for the annotated JDK, \<checker.jar>, and the Type Annotations compiler:
+
+\begin{Verbatim}
+configurations {
+  ... extisting configurations ...
+  checkerFrameworkJavac {
+     description = 'a customization of the Open JDK javac compiler with additional support for type annotations'
+  }
+  checkerFrameworkAnnotatedJDK {
+     description = 'a copy of JDK classes with Checker Framework type qualifers inserted'
+  }
+  checkerFramework {
+     description = 'The Checker Framework: custom pluggable types for Java'
+  } 
+}
+\end{Verbatim}
+
+\item Using the Maven coordinates, declare the Checker Framework dependencies:
+
+\begin{Verbatim}
+dependencies {
+  ... extisting dependencies...
+
+  ext.checkerFrameworkVersion = '1.9.11'
+  checkerFrameworkJavac "org.checkerframework:compiler:${checkerFrameworkVersion}"
+  checkerFrameworkAnnotatedJDK "org.checkerframework:jdk8:${checkerFrameworkVersion}"
+  checkerFramework "org.checkerframework:checker:${checkerFrameworkVersion}"
+  compile "org.checkerframework:checker-qual:${checkerFrameworkVersion}"
+}
+\end{Verbatim}
+
+\item Direct all tasks of type JavaCompile to use the desired checkers:
+
+\begin{Verbatim}
+allprojects {
+  tasks.withType(JavaCompile).all { JavaCompile compile ->
+
+    compile.options.debug = true
+    compile.options.compilerArgs = [
+          '-processor', 'org.checkerframework.checker.nullness.NullnessChecker',
+          '-processorpath', "${configurations.checkerFramework.asPath}",
+          // uncomment to turn Checker Framework errors into warnings
+          //'-Awarns',
+          '-source', '7',
+          '-target', '7',
+          "-Xbootclasspath/p:${configurations.checkerFrameworkAnnotatedJDK.asPath}"
+        ]
     options.fork = true
-    options.forkOptions.executable = "$CHECKERFRAMEWORK/checker/bin/javac"
+    options.forkOptions.jvmArgs += ["-Xbootclasspath/p:${configurations.checkerFrameworkJavac.asPath}"]
   }
 }
 \end{Verbatim}
 
-% extra $ to unconfuse Emacs's LaTeX mode
+Alternatively, create a new task to run the desired checkers:
```

Comment:
Why is this text only in the Java 7 instructions?  Can it be added to the Java 8 instructions too?  (The merging of the Java 7 and Java 8 instructions that I suggested should take care of this.)